pkg/alertmanager/amcfg.go

```diff
@@ -428,6 +428,9 @@ func (cg *configGenerator) convertSlackConfig(ctx context.Context, in monitoring
 			return nil, errors.Errorf("failed to get key %q from secret %q", in.APIURL.Key, in.APIURL.Name)
 		}
 		out.APIURL = strings.TrimSpace(url)
+		if _, err := ValidateURL(out.APIURL); err != nil {
```

Comment:
since there are several places where we do 1) get secret key ref and 2) validate URL, maybe it's worth having a common method? it could also trim spaces as done here (but not at the other places currently).